esrally/driver/runner.py

```diff
@@ -311,6 +324,69 @@ async def __aexit__(self, exc_type, exc_val, exc_tb):
             return False
 
 
+class AssertingRunner(Runner, Delegator):
+    assertions_enabled = False
+
+    def __init__(self, delegate):
+        super().__init__(delegate=delegate)
+        self.predicates = {
+            ">": self.greater_than,
+            ">=": self.greater_than_or_equal,
+            "<": self.smaller_than,
+            "<=": self.smaller_than_or_equal,
+            "==": self.equal,
+        }
+
+    def greater_than(self, expected, actual):
+        return actual > expected
```

Comment:
I notice there are no type restrictions here? i.e. we could in theory compare strings. Is this intentional and left to the user to ensure valid types are being compared?

Reply:
Yes, it's intentional that we're not too restrictive here.

Reply:
👍 maybe we could warn on type comparisons but probably not worth the cost.